addon-test-support/@ember/test-helpers/dom/-is-maxlength-constrained.ts

```diff
@@ -0,0 +1,15 @@
+/**
+  @private
+  @param {Element} element the element to check
+  @returns {boolean} `true` when the element should constrain input by the maxlength attribute, `false` otherwise
+*/
+export default function isMaxLengthConstrained(element: Element) {
+  // ref: https://html.spec.whatwg.org/multipage/input.html#concept-input-apply
+  const constrainedInputTypes = ['text', 'search', 'url', 'tel', 'email', 'password'];
+  return (
+    element.hasAttribute('maxlength') &&
```

Comment:
I'd prefer properties over attributes, like `element.maxLength`. Not sure if it makes the difference here though.

Reply:
I left this as is. Using properties:
`!!Number(element.maxLength)` was giving me lint error: `error TS2339: Property 'maxLength' does not exist on type 'Element'.`, and I got in over my head on casting trying to fix it